python-sdk/src/astro/files/locations/google/gdrive.py

```diff
@@ -0,0 +1,150 @@
+from astro.files.locations.base import BaseFileLocation
+
+
+def _quote(s: str) -> str:
+    s = s.replace("'", "\\'")  # Escape single quote characters.
+    return f"'{s}'"  # Quote.
+
+
+def _find_item_id(path_parts: str, *, conn: Any) -> str:
+    """Drill down the directory hierarchy to find the ID of an item.
+
+    Google Drive does not have a very good random access API by path, so we
+    must look at each directory, level by level, to find the item we want.
+    """
+    current_parent = "root"
+    folders = path_parts.split("/")
+    # First tries to enter directories
+    for current_folder in folders:
+        conditions = [
+            "mimeType = 'application/vnd.google-apps.folder'",
+            f"name='{current_folder}'",
+        ]
+        if current_parent != "root":
+            conditions.append(f"'{current_parent}' in parents")
+        result = (
+            conn.files().list(q=" and ".join(conditions), spaces="drive", fields="files(id, name)").execute()
+        )
+        files = result.get("files", [])
+        if not files:
+            # If the directory does not exist, break loops
+            break
+        current_parent = files[0].get("id")
+    return current_parent
+
+
+def _find_contains(folder_id: str, pattern: str, *, conn: Any) -> Iterator[str]:
+    """Find files in a directory matching given pattern.
+
+    This uses *contains* to search. See Google Drive documentation to
+    understand the implications:
+    https://developers.google.com/drive/api/guides/ref-search-terms
+    """
+    page_token = None
+    while True:
+        response = (
+            conn.files()
+            .list(
+                q=f"name contains {_quote(pattern)} and {_quote(folder_id)} in parents ",
+                fields="nextPageToken, files(name,id, mimeType,webContentLink,size)",
+                pageToken=page_token,
+            )
+            .execute()
+        )
+        yield from (f["webContentLink"] for f in response["files"])
+        page_token = response.get("nextPageToken", None)
+        if page_token is None:
+            return
+
+
+class GdriveLocation(BaseFileLocation):
+    """Handler for Google Drive operators."""
+
+    location_type = FileLocation.GOOGLE_DRIVE
+
+    @property
+    def hook(self) -> GoogleDriveHook:
+        if self.conn_id:
+            return GoogleDriveHook(gcp_conn_id=self.conn_id)
+        return GoogleDriveHook()
+
+    @property
+    def transport_params(self) -> dict:
+        """Get Google Drive client."""
+        return {"client": self.hook.get_conn()}
+
+    @property
+    def openlineage_dataset_namespace(self) -> str:
+        """
+        Returns the open lineage dataset namespace as per
+        https://github.com/OpenLineage/OpenLineage/blob/main/spec/Naming.md
+        """
+        parsed_url = urllib.parse.urlsplit(self.path)
+        return f"{parsed_url.scheme}://{parsed_url.netloc}"
+
+    @property
+    def openlineage_dataset_name(self) -> str:
```

Comment:
We should also add a method for `openlineage_dataset_uri()` as per https://github.com/astronomer/astro-sdk/pull/1306 along with tests

Reply:
I can see only the databases got changed in the PR. Does the change is needed for file/locations as well?

Reply:
Yes we can omit URI for locations